packages/gcs-store/src/locker/GCSLockFile.ts

```diff
@@ -0,0 +1,136 @@
+    this.releaseFile = bucket.file(`${name}.release`)
+  }
+  /**
+   * Create the lockfile with the specified exp time. Throws if the file already exists
+   */
+  public async create(exp: number) {
+    const metadata = {
+      metadata: {exp},
+      // TODO: this does nothing?
+      cacheControl: 'no-store',
+    }
+
+    await this.lockFile.save('', {
+      preconditionOpts: {ifGenerationMatch: 0},
+      metadata,
+    })
+
+    this.currentMetaGeneration = (await this.getMeta()).metageneration
+  }
+
+  /**
+   * Fetch metadata of the lock file.
+   */
+  public async getMeta() {
+    return (await this.lockFile.getMetadata())[0]
+  }
+
+  /**
+   * Refresh our own lockfile. Throws if it does not exist or the file is modified by another instance.
+   */
+  public async refreshOwn(exp: number) {
+    const metadata: FileMetadata = {
+      exp,
+    }
+    const res = await this.lockFile.setMetadata(metadata, {
+      ifMetaGenerationMatch: this.currentMetaGeneration,
+    })
+    this.currentMetaGeneration = res[0].metageneration
+    log('updated currentMetaGeneration', this.currentMetaGeneration)
+  }
+  /**
+   * Check if a release request has been submitted to our own lockfile. Throws if it does not exist or the file is modified by another instance.
+   */
+  public async checkOwnReleaseRequest() {
+    const meta = await this.getMeta()
+    if (meta.metageneration !== this.currentMetaGeneration) {
+      throw new Error('This lockfile has been already taken by another instance.')
+    }
+
+    const releaseRequestExists = (await this.releaseFile.exists())[0]
+    return releaseRequestExists
+  }
+
+  /**
+   * Delete our own lockfile if it still exists.
+   */
+  public async deleteOwn() {
+    try {
+      await this.deleteReleaseRequest()
+      await this.lockFile.delete({ifGenerationMatch: this.currentMetaGeneration})
+    } catch (err) {
+      //Probably already deleted, no need to report
+    }
+  }
+
+  /**
+   * Request releasing the lock from another instance.
+   * As metadata edits are only prohibited for the owner
+   * (so it can keep track of metageneration),
+   * we write to a separate file.
+   */
+  public async requestRelease() {
+    try {
+      await this.releaseFile.save('', {
+        preconditionOpts: {ifGenerationMatch: 0},
+      })
+      log('requestRelease success')
+    } catch (err) {
+      if (err.code === 412) {
+        //Release file already created, no need to report
+        return
+      }
+      log('requestRelease error', err)
+      throw err
+    }
+  }
+
+  /**
+   * Delete the unhealthy file of a previous lock.
+   */
+  public async deleteUnhealthy(metaGeneration: number) {
+    await this.deleteReleaseRequest()
+    await this.lockFile.delete({ifMetagenerationMatch: metaGeneration})
+  }
+
+  /**
+   * Delete release request file of the lock if exists.
+   */
+  protected async deleteReleaseRequest() {
+    try {
+      await this.releaseFile.delete()
+    } catch (err) {}
```

Comment:
Only errors about the object not existing should be ignored. All other errors should be thrown.